src/pypi_attestation_models/_cli.py

```diff
@@ -0,0 +1,294 @@
+if typing.TYPE_CHECKING:
+    from collections.abc import Iterable
+
+logging.basicConfig(format="%(message)s", datefmt="[%X]", handlers=[logging.StreamHandler()])
+_logger = logging.getLogger(__name__)
+_logger.setLevel(logging.INFO)
+
+
+def _parser() -> argparse.ArgumentParser:
+    parent_parser = argparse.ArgumentParser(add_help=False)
+
+    parent_parser.add_argument(
+        "-v",
+        "--verbose",
+        action="count",
+        default=0,
+        help="Run with additional debug logging; supply multiple times to increase verbosity",
+    )
+
+    parser = argparse.ArgumentParser(
+        prog="pypi-attestation-models",
+        description="Sign, inspect or verify PEP 740 attestations",
+        parents=[parent_parser],
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+    )
+
+    parser.add_argument(
+        "-V",
+        "--version",
+        action="version",
+        version=f"pypi-attestation-models {__version__}",
+    )
+
+    subcommands = parser.add_subparsers(
+        required=True,
+        dest="subcommand",
+        metavar="COMMAND",
+        help="The operation to perform",
+    )
+
+    sign_command = subcommands.add_parser(
+        name="sign", help="Sign one or more inputs", parents=[parent_parser]
+    )
+
+    sign_command.add_argument(
+        "--staging",
+        action="store_true",
+        default=False,
+        help="Use the staging environment",
+    )
+
+    sign_command.add_argument(
+        "files",
+        metavar="FILE",
+        type=Path,
+        nargs="+",
+        help="The file to sign",
+    )
+
+    verify_command = subcommands.add_parser(
+        name="verify",
+        help="Verify one or more inputs",
+        parents=[parent_parser],
+    )
+
+    verify_command.add_argument(
+        "--identity",
+        type=str,
+        required=True,
+        help="Signer identity",
+    )
+
+    verify_command.add_argument(
+        "--staging",
+        action="store_true",
+        default=False,
+        help="Use the staging environment",
+    )
+
+    verify_command.add_argument(
+        "files",
+        metavar="FILE",
+        type=Path,
+        nargs="+",
+        help="The file to sign",
+    )
+
+    inspect_command = subcommands.add_parser(
+        name="inspect",
+        help="Inspect one or more inputs",
+        parents=[parent_parser],
+    )
+
+    inspect_command.add_argument(
+        "--dump-bytes",
+        action="store_true",
+        default=False,
+        help="Dump the bytes of the signature",
+    )
+
+    inspect_command.add_argument(
+        "files",
+        metavar="FILE",
+        type=Path,
+        nargs="+",
+        help="The file to sign",
+    )
+
+    return parser
+
+
+def _die(message: str) -> NoReturn:
+    """Handle errors and terminate the program with an error code."""
+    _logger.error(message)
+    raise SystemExit(1)
+
+
+def _validate_files(files: list[Path] | Iterable[Path], should_exists: bool = True) -> None:
```

Comment:
`list[Path]` should be subtyped under `Iterable[Path]`, so this should be fine:

```suggestion
def _validate_files(files: Iterable[Path], should_exists: bool = True) -> None:
```